camel-idea-plugin/src/main/java/org/apache/camel/idea/annotator/CamelAnnotator.java

```diff
@@ -0,0 +1,220 @@
+ * or simple expression to highlight the error on the editor
+ */
+public class CamelAnnotator implements Annotator {
+    @Override
+    public void annotate(@NotNull PsiElement element, @NotNull AnnotationHolder holder) {
+        if (ServiceManager.getService(element.getProject(), CamelService.class).isCamelPresent()) {
+            String fromElement = IdeaUtils.extractTextFromElement(element);
+            if (fromElement != null && !fromElement.isEmpty() && !fromElement.trim().endsWith("&") && !fromElement.trim().endsWith("?")
+                && !fromElement.trim().endsWith("{{") && !fromElement.trim().endsWith("=")
+                && !fromElement.trim().endsWith("${")) {
+                validateEndpointProperties(element, holder, fromElement);
+                //Comment out until we solve the classloader issue.
+                //validateSimpleExpression(element, holder, fromElement);
+            }
+        }
+    }
+
+    /**
+     * Validate endpoint options list aka properties. eg "timer:trigger?delay=1000&bridgeErrorHandler=true"
+     * if the URI is not valid a error annotation is created and highlight the invalid value.
+     */
+    private void validateEndpointProperties(@NotNull PsiElement element, @NotNull AnnotationHolder holder, String fromElement) {
+        if ((fromElement.contains("?") || fromElement.contains("&")) && IdeaUtils.isConsumerEndpoint(element)) {
+            CamelCatalog catalogService = ServiceManager.getService(element.getProject(), CamelCatalogService.class).get();
+            EndpointValidationResult validateEndpointProperties = catalogService.validateEndpointProperties(fromElement);
+            extractMapValue(validateEndpointProperties, validateEndpointProperties.getInvalidBoolean(), fromElement, element, holder);
+            extractMapValue(validateEndpointProperties, validateEndpointProperties.getInvalidEnum(), fromElement, element, holder);
+            extractMapValue(validateEndpointProperties, validateEndpointProperties.getInvalidInteger(), fromElement, element, holder);
+            extractMapValue(validateEndpointProperties, validateEndpointProperties.getInvalidNumber(), fromElement, element, holder);
+            extractMapValue(validateEndpointProperties, validateEndpointProperties.getInvalidReference(), fromElement, element, holder);
+        }
+    }
+
+    // TODO : Solve classloader issue class not found when CamelCatalog access a camel core java file.
+    // The problem is releated to the CamelCatalog is using the plugin classloader, but the Camel core
+    // exits on the project/module classloader
+    /**
+     * Validate simple expression. eg "simple("${body}"
+     * if the expression is not valid a error annotation is created and highlight the invalid value.
+     */
+    private void validateSimpleExpression(@NotNull PsiElement element, @NotNull AnnotationHolder holder, String fromElement) {
```

Comment:
I would move simple validation out of this class into its own CamelSimpleAnnotator so they are separated.